src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -219,4 +233,132 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
+     * message is parsed for its slf4j context (logger name, logger level etc.)
+     * and the specified {@code BiConsumer} is invoked with the log message and
+     * slf4j context.
+     * 
+     * <p>Use this method only if you want a behavior that differentiates from the
+     * default behavior as enabled by method
+     * {@link #enableDefaultSlf4jLoggingOfOutputData()}.
+     * 
+     * <p>NOTE: You might want to call method {@link #clearOutPutStreams()} before
+     * calling this method.
+     * 
+     * <p>implementation detail:
+     * <ul>
+     * <li>if log message begins with {@code [debug]} then log level is set to
+     * {@code DEBUG}, otherwise log level is {@code INFO}.</li>
+     * <li>the appium sub module name is parsed from the log message and used as
+     * logger name (prefixed with "appium.service.", all lower case, spaces
+     * removed). If no appium sub module is detected then "appium.service" is
+     * used as logger name.</li>
+     * </ul>
+     * Example log-message: "[ADB] Cannot read version codes of " is logged by
+     * {@code appium.service.adb} at level {@code INFO} <br>
+     * Example log-message: "[debug] [XCUITest] Xcode version set to 'x.y.z' "
+     * is logged by {@code appium.service.xcuitest} at level {@code DEBUG}
+     * <br>
+     * 
+     * @param slf4jLogMessageConsumer
+     *            BiConsumer block to be executed when a log message is
+     *            available.
+     */
+    public void addSlf4jLogMessageConsumer(
+            BiConsumer<String, Slf4jLogMessageContext> slf4jLogMessageConsumer) {
+        addLogMessageConsumer(logMessage -> {
+            slf4jLogMessageConsumer.accept(logMessage, parseSlf4jContextFromLogMessage(logMessage));
+        });
+    }
+
+    static Slf4jLogMessageContext parseSlf4jContextFromLogMessage(String logMessage) {
+        Matcher m = LOGGER_CONTEXT_PATTERN.matcher(logMessage);
+        String loggerName = APPIUM_SERVICE_SLF4J_LOGGER_PREFIX;
+        Level level = INFO;
+        if (m.find()) {
+            loggerName += "." + m.group(2).toLowerCase().replaceAll("\\s+", "");
+            if (m.group(1) != null) {
+                level = DEBUG;
+            }
+        }
+        return new Slf4jLogMessageContext(loggerName, level);
+    }
+
+    /**
+     * When a complete log message is available (from server output data), the
+     * specified {@code Consumer} is invoked with that log message.
+     * 
+     * <p>NOTE: You might want to call method {@link #clearOutPutStreams()} before
+     * calling this method.
+     * 
+     * <p>If the Consumer fails and throws an exception the exception is logged (at
+     * WARN level) and execution continues.
+     * <br>
+     * 
+     * @param consumer
+     *            Consumer block to be executed when a log message is available.
+     * 
+     */
+    public void addLogMessageConsumer(Consumer<String> consumer) {
+        addOutPutStream(new OutputStream() {
```

Comment:
Output

Reply:
I'm not following this comment? `addOutPutStream(...)` is the name of an already existing method.

Reply:
There is a typo in this word

Reply:
@mykola-mokhnach 
I'm still not following what this has to do with this PR?
As I said the `addOutPutStream(...)` method is already existing. Changing its name would be a breaking change.

Reply:
then yes, it is better to keep it as is if it was there